src/test/java/io/supertokens/test/userIdMapping/UserIdMappingTest.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void testDuplicateUserIdMapping() throws Exception {
+        String[] args = { "../" };
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        // create a user
+        UserInfo userInfo = EmailPassword.signUp(process.main, "test@example.com", "testPassword");
+
+        String externalUserId = "external-test";
+
+        UserIdMapping.createUserIdMapping(process.main, userInfo.id, externalUserId, null);
+
+        {
+            // duplicate exception with both supertokensUserId and externalUserId
+            Exception error = null;
+            try {
+                UserIdMapping.createUserIdMapping(process.main, userInfo.id, externalUserId, null);
+            } catch (Exception e) {
+                error = e;
+            }
+
+            assertNotNull(error);
+            assertTrue(error instanceof UserIdMappingAlreadyExistsException);
+
+            UserIdMappingAlreadyExistsException usersIdMappingExistsError = (UserIdMappingAlreadyExistsException) error;
+            assertTrue(usersIdMappingExistsError.doesExternalUserIdExist);
+            assertTrue(usersIdMappingExistsError.doesSuperTokensUserIdExist);
+        }
+
+        {
+            // duplicate exception with superTokensUserId
+            Exception error = null;
+            try {
+                UserIdMapping.createUserIdMapping(process.main, userInfo.id, "newExternalId", null);
+            } catch (Exception e) {
+                error = e;
+            }
+
+            assertNotNull(error);
+            assertTrue(error instanceof UserIdMappingAlreadyExistsException);
+
+            UserIdMappingAlreadyExistsException usersIdMappingExistsError = (UserIdMappingAlreadyExistsException) error;
+            assertFalse(usersIdMappingExistsError.doesExternalUserIdExist);
+            assertTrue(usersIdMappingExistsError.doesSuperTokensUserIdExist);
+
+        }
+
+        {
+            // duplicate exception with externalUserId
+
+            UserInfo newUser = EmailPassword.signUp(process.main, "test2@example.com", "testPass123");
+            Exception error = null;
+            try {
+                UserIdMapping.createUserIdMapping(process.main, newUser.id, externalUserId, null);
+            } catch (Exception e) {
+                error = e;
+            }
+
+            assertNotNull(error);
+            assertTrue(error instanceof UserIdMappingAlreadyExistsException);
+
+            UserIdMappingAlreadyExistsException usersIdMappingExistsError = (UserIdMappingAlreadyExistsException) error;
+            assertTrue(usersIdMappingExistsError.doesExternalUserIdExist);
+            assertFalse(usersIdMappingExistsError.doesSuperTokensUserIdExist);
+
+        }
```

Comment:
You should also add a test where you give an unknown st user ID and an already mapped external user ID, and then it should throw UnkownSTUserId error